File: example/example.api/src/main/java/cn/jmicro/example/api/rpc/ISimpleRpc.java
package cn.jmicro.example.api.rpc;

import cn.jmicro.api.annotation.Service;
import cn.jmicro.api.async.IPromise;
import cn.jmicro.api.test.Person;
import cn.jmicro.codegenerator.AsyncClientProxy;

@AsyncClientProxy
@Service(namespace="simpleRpc", version="0.0.1")
public interface ISimpleRpc {
	
	String hello(String name);
	
	String hi(Person p);
	
	 IPromise<String> linkRpc(String msg);
	
	IPromise<String> linkRpcAs(String msg);
}

File: mng.web/src/rpcservice/ISimpleRpc.js
import rpc from "@/rpc/rpcbase";

export default {

  hello: function (msg) {
    let req = rpc.creq(this.sn,this.ns,this.v, 'hello', [msg])
    return rpc.callRpc(req);
  },

  sn:'cn.expjmicro.example.api.rpc.ISimpleRpc',
  ns : 'exampleProvider',
  v:'0.0.1',

}
